tfhe/benches/core_crypto/pbs128_bench.rs

```diff
@@ -0,0 +1,175 @@
+        let message_modulus: Scalar = 1 << 4;
+
+        // Our input message
+        let input_message: Scalar = 3;
+
+        // Delta used to encode 4 bits of message + a bit of padding on Scalar
+        let delta: Scalar = (1 << (Scalar::BITS - 1)) / message_modulus;
+
+        // Apply our encoding
+        let plaintext = Plaintext(input_message * delta);
+
+        // Allocate a new LweCiphertext and encrypt our plaintext
+        let lwe_ciphertext_in: LweCiphertextOwned<Scalar> = allocate_and_encrypt_new_lwe_ciphertext(
+            &small_lwe_sk,
+            plaintext,
+            lwe_modular_std_dev,
+            &mut encryption_generator,
+        );
+
+        // Now we will use a PBS to compute a multiplication by 2, it is NOT the recommended way of
+        // doing this operation in terms of performance as it's much more costly than a
+        // multiplication with a cleartext, however it resets the noise in a ciphertext to a
+        // nominal level and allows to evaluate arbitrary functions so depending on your use
+        // case it can be a better fit.
+
+        // Here we will define a helper function to generate an accumulator for a PBS
+        fn generate_accumulator<F>(
+            polynomial_size: PolynomialSize,
+            glwe_size: GlweSize,
+            message_modulus: usize,
+            delta: Scalar,
+            f: F,
+        ) -> GlweCiphertextOwned<Scalar>
+        where
+            F: Fn(Scalar) -> Scalar,
+        {
+            // N/(p/2) = size of each block, to correct noise from the input we introduce the notion
+            // of box, which manages redundancy to yield a denoised value for several
+            // noisy values around a true input value.
+            let box_size = polynomial_size.0 / message_modulus;
+
+            // Create the accumulator
+            let mut accumulator_scalar: Vec<Scalar> = vec![0; polynomial_size.0];
+
+            // Fill each box with the encoded denoised value
+            for i in 0..message_modulus {
+                let index = i * box_size;
+                accumulator_scalar[index..index + box_size]
+                    .iter_mut()
+                    .for_each(|a| *a = f(i as Scalar) * delta);
+            }
+
+            let half_box_size = box_size / 2;
+
+            // Negate the first half_box_size coefficients to manage negacyclicity and rotate
+            for a_i in accumulator_scalar[0..half_box_size].iter_mut() {
+                *a_i = (*a_i).wrapping_neg();
+            }
+
+            // Rotate the accumulator
+            accumulator_scalar.rotate_left(half_box_size);
+
+            let accumulator_plaintext = PlaintextList::from_container(accumulator_scalar);
+
+            allocate_and_trivially_encrypt_new_glwe_ciphertext(glwe_size, &accumulator_plaintext)
+        }
+
+        let f = |x: Scalar| (x * x + x) % 16;
+        let accumulator: GlweCiphertextOwned<Scalar> = generate_accumulator(
```

Comment:
I think you can jsut use an empty GlweCiphertext and remove the generate accumulator code

Reply:
don't we need the accumulator to be filled for the bootstrap to work?

Reply:
(we skip the external product if an LWE mask element is zero, so as long as the lwe is non trivial we are executing the external product part)

Reply:
ah, right